### PR TITLE
RedeployStaking

### PR DESCRIPTION
**Pull request by passandscore**

### CODE REVIEW

**Comment by passandscore** [comment]
The problem was that once we approved the SF contract and transferred the stake of the child to our USDCx wallet we were trying to send the reward to the child. The sender was the staking contract but the Allocate tokens were initially minted to an owner wallet. The staking contract did not have approval to send on behave of the owner wallet. The way around this was to allow the child to call the mint function. That function will provide the reward.

File: contracts/StakingToken.sol
Comment (acaldas) [inline contracts/StakingToken.sol:101]: We need to remove the minting of the full supply to the owner on line 44, no?

Comment (passandscore) [reply contracts/StakingToken.sol:101]: Well, it is standard that the address that deploys the contract, is the owner of the initially minted tokens. 
https://docs.openzeppelin.com/contracts/4.x/erc20

Perhaps we could mint the tokens to the contract address. Is this what you are thinking? I still like using the mint function in this manner as we will never have to worry about reaching our max on total supply.

Comment (acaldas) [reply contracts/StakingToken.sol:101]: But we are minting the total supply when deploying, doesn’t that mean that the following mints from staking will exceed the total supply?

Comment (acaldas) [reply contracts/StakingToken.sol:101]: Ah I think I get it, totalSupply means “initial supply” not “max supply” right?

Comment (passandscore) [reply contracts/StakingToken.sol:101]: Thats right. There is no cap

**Review by acaldas — APPROVED** [review]

**Pull request merged**
